File: 744.寻找比目标字母大的最小字母.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int i = 0, j = letters.size() - 1;
        while(i <= j){
            int mid = i + (j - i) / 2;
            if(letters[mid] <= target)
                i = mid + 1;
            else
                j = mid - 1;
        }
        return i < letters.size() ? letters[i] : letters[0];
    }
};

File: 二分查找/744.寻找比目标字母大的最小字母.js
/**
 * @param {character[]} letters
 * @param {character} target
 * @return {character}
 */
var nextGreatestLetter = function (letters, target) {
    let l = 0,
        r = letters.length - 1;
    while (l < r) {
        let mid = Math.floor((r - l) / 2) + l;
        if (letters[mid] <= target) {
            l = mid + 1;
        } else {
            r = mid;
        }
    }
    return letters[l] > target ? letters[l] : letters[0];
};